x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolverTests.java

```diff
@@ -133,6 +133,31 @@ public void testMetaFieldsAreIgnored() throws Exception {
         assertEquals(DataType.KEYWORD, esIndex.mapping().get("text").getDataType());
     }
 
+    public void testMergeIncompatibleCapabilitiesOfObjectFields() throws Exception {
+        Map<String, Map<String, FieldCapabilities>> fieldCaps = new HashMap<>();
+        // define a sub-field
+        addFieldCaps(fieldCaps, "a.keyword", "keyword", true, true);
+
+        Map<String, FieldCapabilities> multi = new HashMap<>();
+        multi.put("long", new FieldCapabilities("a", "long", true, true, new String[] { "one-index" }, null, null));
```

Comment:
I wouldn't name the field the same as the type of the field. If we screw up something in the future and somehow the type of the field instead of the name gets logged in the error message, we miss a bug.